WithRefs type strips null from nullable fields

### Description

The `WithRefs<TData>` type in `@hey-api/client-nuxt` uses `NonNullable<TData[K]>` which incorrectly strips `null` from nullable field types. This makes it impossible to pass `null` values to API endpoints that expect nullable fields.

## Reproduction

### OpenAPI Schema (backend)
```yaml
components:
  schemas:
    CustomerCreateRequest:
      type: object
      required:
        - name
        - email
        - country
        - street    # required but nullable
        - city      # required but nullable
        - postalCode # required but nullable
      properties:
        name:
          type: string
        email:
          type: string
        country:
          type: string
        street:
          type: string
          nullable: true
        city:
          type: string
          nullable: true
        postalCode:
          type: string
          nullable: true
```

### Generated types.gen.ts
```typescript
export type CustomerCreateRequest = {
    name: string;
    email: string;
    street: null | string;   // Correctly includes null
    city: null | string;     // Correctly includes null
    postalCode: null | string; // Correctly includes null
    country: string;
};
```

### Generated client/types.gen.ts - The Problem
```typescript
type WithRefs<TData> = {
  [K in keyof TData]: NonNullable<TData[K]> extends object
    ? WithRefs<NonNullable<TData[K]>> | Ref<NonNullable<TData[K]>>
    : NonNullable<TData[K]> | Ref<NonNullable<TData[K]>>;  // <-- BUG: NonNullable strips null
};
```

For `street: null | string`, this resolves to:
- `NonNullable<null | string>` = `string`
- Result: `string | Ref<string>` (null is lost!)

### Usage that fails
```typescript
import type { CustomerCreateRequest } from '~/generated-api'
import { customersCreate } from '~/generated-api'

async function createCustomer(street?: string) {
  await customersCreate({
    composable: '$fetch',
    body: {
      name: 'John',
      email: 'john@example.com',
      country: 'US',
      street: street ?? null,  // ERROR: Type 'string | null' is not assignable to type 'string | Ref<string>'
      city: null,              // ERROR: Type 'null' is not assignable to type 'string | Ref<string>'
      postalCode: null,        // ERROR: Type 'null' is not assignable to type 'string | Ref<string>'
    },
  })
}
```

### TypeScript Error
```
error TS2322: Type 'string | null' is not assignable to type 'string | Ref<string, string>'.
  Type 'null' is not assignable to type 'string | Ref<string, string>'.
```

## Expected Behavior
The `WithRefs` type should preserve `null` in the union:
```typescript
// For street: null | string, it should resolve to:
street: null | string | Ref<string>
```

## Suggested Fix
```typescript
type WithRefs<TData> = {
  [K in keyof TData]: NonNullable<TData[K]> extends object
    ? WithRefs<NonNullable<TData[K]>> | Ref<NonNullable<TData[K]>> | Extract<TData[K], null>
    : NonNullable<TData[K]> | Ref<NonNullable<TData[K]>> | Extract<TData[K], null>;
};
```

Or more simply, preserve the original null:
```typescript
type WithRefs<TData> = {
  [K in keyof TData]: null extends TData[K]
    ? (NonNullable<TData[K]> | Ref<NonNullable<TData[K]>> | null)
    : (NonNullable<TData[K]> extends object
        ? WithRefs<NonNullable<TData[K]>> | Ref<NonNullable<TData[K]>>
        : NonNullable<TData[K]> | Ref<NonNullable<TData[K]>>);
};
```

## Workaround
Currently using `as never` type assertion to bypass:
```typescript
const body: CustomerCreateRequest = {
  name: 'John',
  email: 'john@example.com',
  country: 'US',
  street: null,
  city: null,
  postalCode: null,
}
await customersCreate({
  composable: '$fetch',
  body: body as never,  // Workaround
})
```

## Environment
- @hey-api/openapi-ts: 0.89.2
- TypeScript: 5.9.3
- Nuxt: 4.x


### Reproducible example or configuration

https://stackblitz.com/edit/hey-api-client-fetch-example

### OpenAPI specification (optional)

_No response_

### System information (optional)

_No response_

--- a/.changeset/thin-pugs-provide.md
+++ b/.changeset/thin-pugs-provide.md
@@ -1,0 +1,5 @@
+---
+"@hey-api/openapi-ts": patch
+---
+
+**plugin(@hey-api/client-nuxt)**: preserve null in `WithRefs` type for nullable fields

--- a/packages/openapi-ts/src/plugins/@hey-api/client-nuxt/bundle/types.ts
+++ b/packages/openapi-ts/src/plugins/@hey-api/client-nuxt/bundle/types.ts
@@ -74,15 +74,8 @@ export interface RequestOptions<
   Url extends string = string,
 > extends Config,
     WithRefs<{
-      /**
-       * Any body that you want to add to your request.
-       *
-       * {@link https://developer.mozilla.org/docs/Web/API/fetch#body}
-       */
-      body?: unknown;
       path?: FetchOptions<unknown>['query'];
       query?: FetchOptions<unknown>['query'];
-      rawBody?: unknown;
     }>,
     Pick<
       ServerSentEventsOptions<ResT>,
@@ -93,8 +86,15 @@ export interface RequestOptions<
       | 'sseMaxRetryDelay'
     > {
   asyncDataOptions?: AsyncDataOptions<ResT, ResT, KeysOf<ResT>, DefaultT>;
+  /**
+   * Any body that you want to add to your request.
+   *
+   * {@link https://developer.mozilla.org/docs/Web/API/fetch#body}
+   */
+  body?: NonNullable<unknown> | Ref<NonNullable<unknown>> | null;
   composable?: TComposable;
   key?: string;
+  rawBody?: NonNullable<unknown> | Ref<NonNullable<unknown>> | null;
   /**
    * Security mechanism(s) to use for the request.
    */

--- a/packages/openapi-ts/src/plugins/@hey-api/client-nuxt/bundle/types.ts
+++ b/packages/openapi-ts/src/plugins/@hey-api/client-nuxt/bundle/types.ts
@@ -31,8 +31,14 @@ export type QuerySerializer = (
 
 type WithRefs<TData> = {
   [K in keyof TData]: NonNullable<TData[K]> extends object
-    ? WithRefs<NonNullable<TData[K]>> | Ref<NonNullable<TData[K]>>
-    : NonNullable<TData[K]> | Ref<NonNullable<TData[K]>>;
+    ?
+        | WithRefs<NonNullable<TData[K]>>
+        | Ref<NonNullable<TData[K]>>
+        | Extract<TData[K], null>
+    :
+        | NonNullable<TData[K]>
+        | Ref<NonNullable<TData[K]>>
+        | Extract<TData[K], null>;
 };
 
 // copied from Nuxt

